Add linting for Javascript files

kolipka opened the pull request:
lint should be run before each commit.

IoT.js-DCO-1.0-Signed-off-by: Konrad Lipner k.lipner@samsung.com

yichoi commented on .eslintrc.js:
```diff
@@ -0,0 +1,36 @@
+/* global module */
+/*eslint quotes: [2, "double"]*/
```
please add space. It should be ``/* eslint qutes: [2, "double"] */``

yichoi commented:
travis failed because of tidy error for js files.
I think js file should be fixed first along with js coding convention before this patch landed

nova0821 commented:
I think this should be included in travis test. 

Also, We have the rules for JS code in https://github.com/Samsung/iotjs/wiki/Coding-Style-Guideline#coding-style-guideline-for-javascript.

Does it follow these rules?

kolipka commented:
@yichoi it was tidy check. I added ESLint config file to ignored, added empty line at the end of files and keep 80 chars line limit.
I would like to add this first, and if we agree on rules at overall, I will make next PR fixing all errors and adding this test to travis.

About all rules that are enabled you can read here: http://eslint.org/docs/rules/
in .eslintrc.js you can check rules that I've changed

@nova0821 I think it should follow our rules, but I will double check.

hs0225 commented on tools/lint.sh:
```diff
@@ -0,0 +1,34 @@
+
+if ! [ -x "$IOTPATH/node_modules/.bin/eslint" ] ; then
+  echo "ESLint not installed!\nYou can install it using:"
+  echo "$IOTPATH$$npm install"
```
To run the command `npm install`, please create `package.json` file.

yichoi commented:
Ok. please check above comment of mine.

kolipka commented:
Travis failed again - this time because 80 chars limit was exceeded after eslint did automated fixes. 
Are we sure we want to keep such limit? I would suggest to extend it at least to 120 chars per line.

yichoi commented:
@kolipka I will not extend to 120 per line since I respect what the committer or contributor have been here even it is not his/her favorites. Please keep following the [coding rules](https://github.com/Samsung/iotjs/wiki/Coding-Style-Guideline#coding-style-guideline-for-javascript) @nova0821 mentioned before.

LaszloLango commented on src/js/ble_hci_socket_gap.js:
```diff
@@ -41,7 +41,7 @@ var util = require('util');
 var Hci = require('ble_hci_socket_hci');
 
-var isLinux = (process.platform === 'linux');
+// TODO: var isLinux = (process.platform === 'linux');
```
What is the TODO here? Please be more specific.

kolipka replied:
isLinux is unused - I could delete it, but below are commented out lines that use it. Unfortunatelly whole BLE looks like WIP and in my personal opinion should not be commited. I want to avoid doing harm removing something useful for its author, that is why I just commented it out, with TODO marker.

kolipka commented:
I started conversation in https://github.com/Samsung/iotjs/issues/631 about js code style

kolipka commented:
This PR is missleading. After #631 and #639 will be settled, we can fix errors and add it to Travis.
